force-app/main/default/classes/CON_PreferredPhone_TEST.cls

```diff
@@ -194,87 +208,137 @@ private class CON_PreferredPhone_TEST {
     static void testBatchOther() {
         disablePreferredPhone();
         
-        Contact newCont = new Contact(
+        insert new Contact(
             FirstName = 'JohnnyTest1',
             LastName = 'JohnnyTest',
             Phone  = '111111111',
             OtherPhone = '111111111'
         );
-        insert newCont;
         
         Test.startTest();
             enablePreferredPhone();
             CON_Email_BATCH batch = new CON_Email_BATCH(null);
             Database.executeBatch(batch);
         Test.stopTest();
 
-        Contact newContAfterList = [SELECT Id, FirstName, LastName, Name, PreferredPhone__c FROM Contact];
-                
-        System.assertEquals('Other Phone', newContAfterList.PreferredPhone__c);
+        System.assertEquals('Other Phone', getCurrentContact().PreferredPhone__c);
        
     }   
     
-    /*  Validation when there are more than one phone fields but not Preferred Phone */
+    /* Validation when there are more than one phone fields but not Preferred Phone 
+     * Test Scenario: T-3903371
+     */
     @isTest 
-    static void testErrorPrefBlank() {
+    static void testErrorPrefPhoneAmbiguous() {
         enablePreferredPhone();
+        Contact newCont = new Contact(
+            FirstName = 'JohnnyTest1',
+            LastName = 'JohnnyTest',
+            WorkPhone__c  = '111111111',
+            OtherPhone = '0000000000'
+        );
 
-        try {
-            Contact newCont = new Contact(
-                FirstName = 'JohnnyTest1',
-                LastName = 'JohnnyTest',
-                WorkPhone__c  = '111111111',
-                OtherPhone = '0000000000'
-            );
-            
+        try {    
             Test.startTest();
             insert newCont;
             Test.stopTest();
-            
+            System.assert(false, 'Multiple phones without preference specified should error out - when enforced');
```

Comment:
No exception situation is checked here, great! There should be similar checks in other places